### DISCUSSION TITLE
OpenRoad db query

### DISCUSSION THREAD

**Discussion by CathedralsOfSand** [The-OpenROAD-Project/OpenROAD#5451] [Q&A] [answered]
How do you dump detailed cell usage statistics from a finished database?

I have the cell stats from the yosys synth run but want similar data from the timing closed routed db. 

Thanks

**Comment by maliberty** [comment]
`report_cell_usage` is the simplest

**Reply by CathedralsOfSand** [reply]
Thanks Matt. I knew about that one but it doesn't list the count of each std cell... nand2, nor1, buf etc. I can see the information in the def file but was hoping there was a way to write it out like the yosys script does.

**Reply by maliberty** [reply]
You could loop over the instances in tcl and collect it yourself.  We could add a -verbose option.  Do you just want a count of the number of instances per LEF master?

**Comment by maliberty** [comment]
https://github.com/The-OpenROAD-Project/OpenROAD/pull/5455 is quick stab at this.  What do you think of https://github.com/The-OpenROAD-Project/OpenROAD/pull/5455/files#diff-2da4a6ba84e5e60ce2a822063c4acd592ca2c6fc35f81f3d3d966f78ba7e1675 ?

**Reply by CathedralsOfSand** [reply]
Looks good, thanks. How do I install it?

**Reply by maliberty** [reply]
I'll merge it and then you can update.

**Reply by CathedralsOfSand** [reply]
Okay, guess I'm showing how new to OpenRoad I am. How do I do an update? Thanks for your help Matt. I have gained a deep appreciation for what you and the OpenRoad team are doing. Thanks.

**Reply by maliberty** [reply]
It depends on what method you used.  If you built from source then you need to use the latest code.  If you are using a docker image or pre-built binary it should be available by snow.

**Reply by CathedralsOfSand** [reply]
I used local source. How do I update? Can I just update the couple of files you touched and then what? Do I need to recompile the whole thing or is there a short cut?

Thanks again for your help.

**Reply by QuantamHD** [reply]
If your on the master branch just run `git pull` and it'll update the code to the latest version. Then just rebuild as normal

**Reply by maliberty** [reply]
If you are using ORFS's build_openroad.sh then add --latest

**Reply by CathedralsOfSand** [reply]
Got thank you both.

**Reply by CathedralsOfSand** [reply]
Updated to latest, works great! Thanks Matt.

**Reply by CathedralsOfSand** [reply]
Matt when I updated to --latest it seems to have broken some flows. It doesn't see the abc command. Any thoughts?

**Reply by maliberty** [reply, marked as answer]
There is no abc command in OR (nor was there) - that's a yosys command.
